examples/FisherPrice_cu/cufisher_price_v2.cu

```diff
@@ -0,0 +1,231 @@
+                                 Queue_t *queue)
+{
+  int particle_index;
+  for (int i = blockIdx.x * blockDim.x + threadIdx.x; i < n; i += blockDim.x * gridDim.x) {
+    if (!queue->dequeue(particle_index)) return;
+
+    // check if the particle is still alive (E>0)
+    if ((*block)[particle_index].energy == 0) return;
+
+    // pair production
+    float eloss = 0.5f * (*block)[particle_index].energy;
+    (*block)[particle_index].energy -= eloss;
+
+    // here I need to create a new particle
+    auto secondary_track = block->NextElement();
+    assert(secondary_track != nullptr && "No slot available for secondary track");
+    secondary_track->energy = eloss;
+
+    // increase the counter of secondaries
+    scor->secondaries.fetch_add(1);
+  }
+}
+
+/* this GPU kernel function is used to initialize the random states */
+__global__ void init(curandState_t *states)
+{
+  /* we have to initialize the state */
+  curand_init(0, 0, 0, states);
+}
+
+//
+
+int main()
+{
+
+  curandState_t *state;
+  cudaMalloc((void **)&state, sizeof(curandState_t));
+  init<<<1, 1>>>(state);
+  cudaDeviceSynchronize();
+
+  // Capacity of the different containers
+  constexpr int capacity = 1 << 20;
+
+  using Queue_t = adept::mpmc_bounded_queue<int>;
+
+  constexpr int numberOfProcesses = 3;
+  char *buffer[numberOfProcesses];
+
+  Queue_t **queues = nullptr;
+  cudaMallocManaged(&queues, numberOfProcesses * sizeof(Queue_t *));
+
+  size_t buffersize = Queue_t::SizeOfInstance(capacity);
+
+  for (int i = 0; i < numberOfProcesses; i++) {
+    buffer[i] = nullptr;
+    cudaMallocManaged(&buffer[i], buffersize);
+
+    queues[i] = Queue_t::MakeInstanceAt(capacity, buffer[i]);
+  }
+
+  // Allocate the content of Scoring in a buffer
+  char *buffer_scor = nullptr;
+  cudaMallocManaged(&buffer_scor, sizeof(Scoring));
+  Scoring *scor = Scoring::MakeInstanceAt(buffer_scor);
+  // Initialize scoring
+  scor->secondaries     = 0;
+  scor->totalEnergyLoss = 0;
+
+  // Allocate a block of tracks with capacity larger than the total number of spawned threads
+  // Note that if we want to allocate several consecutive block in a buffer, we have to use
+  // Block_t::SizeOfAlignAware rather than SizeOfInstance to get the space needed per block
+
+  using Block_t    = adept::BlockData<MyTrack>;
+  size_t blocksize = Block_t::SizeOfInstance(capacity);
+  char *buffer2    = nullptr;
+  cudaMallocManaged(&buffer2, blocksize);
+  auto block = Block_t::MakeInstanceAt(capacity, buffer2);
+
+  // initializing one track in the block
+  auto track    = block->NextElement();
+  track->energy = 100.0f;
+
+  // initializing second track in the block
+  auto track2    = block->NextElement();
+  track2->energy = 30.0f;
+
+  //
+
+  const int num_streams = 8;
+  cudaStream_t streams[num_streams];
+
+  cudaStreamCreate(&streams[0]);
+  cudaStreamCreate(&streams[1]);
+  cudaStreamCreate(&streams[2]);
```

Comment:
Should these be commented out, or the version using streams be in a "v3" example?

Reply:
Yes, a left over from a different 'experiment'. Removed for now.